### PR TITLE
Add infra/gcp/ensure-audit-perms.sh

### PR DESCRIPTION
**Pull request by hh**
/assign @spiffxp

I think this is still a manual run, but at least it's now documented.

I used the following to get the current perms:

```
gcloud organizations get-iam-policy 758905017065 \
  | yq -c -r \
  '.bindings[] | select(.members[] | contains("group:k8s-infra-gcp-auditors@kubernetes.io")) | .role'
```

```
organizations/758905017065/roles/StorageBucketLister
roles/compute.viewer
roles/dns.reader
roles/iam.securityReviewer
roles/resourcemanager.organizationViewer
roles/serviceusage.serviceUsageConsumer
```

It will run the following:
```
gcloud organization add-iam-policy-binding --member=group:k8s-infra-gcp-auditor@kubernetes.io \
--role=organizations/758905017065/roles/StorageBucketLister
gcloud organization add-iam-policy-binding --member=group:k8s-infra-gcp-auditor@kubernetes.io \
--role=roles/compute.viewer
gcloud organization add-iam-policy-binding --member=group:k8s-infra-gcp-auditor@kubernetes.io \
--role=roles/dns.reader
gcloud organization add-iam-policy-binding --member=group:k8s-infra-gcp-auditor@kubernetes.io \
--role=roles/iam.securityReviewer
gcloud organization add-iam-policy-binding --member=group:k8s-infra-gcp-auditor@kubernetes.io \
--role=roles/resourcemanager.organizationViewer
gcloud organization add-iam-policy-binding --member=group:k8s-infra-gcp-auditor@kubernetes.io \
--role=roles/secretmanager.viewer
gcloud organization add-iam-policy-binding --member=group:k8s-infra-gcp-auditor@kubernetes.io \
--role=roles/serviceusage.serviceUsageConsumer
```

### CODE REVIEW

File: infra/gcp/ensure-audit-perms.sh
Comment (spiffxp) [inline infra/gcp/ensure-audit-perms.sh:32]: this looks like leftover copy-pasta from ensure-staging-storage.sh, please remove or update instructions to be relevant to this script

File: infra/gcp/ensure-audit-perms.sh
Comment (spiffxp) [inline infra/gcp/ensure-audit-perms.sh:49]:
```suggestion
    gcloud organizations add-iam-policy-binding \
           --member="${GROUP} \
           --role="${ROLE}"
           "${ORG}"
```

File: infra/gcp/ensure-audit-perms.sh
Comment (spiffxp) [inline infra/gcp/ensure-audit-perms.sh:36]:
```suggestion
ORG="${GCP_ORG}
GROUP="group:k8s-infra-gcp-auditor@kubernetes.io"
```

File: infra/gcp/ensure-audit-perms.sh
Comment (spiffxp) [inline infra/gcp/ensure-audit-perms.sh:1]: This file needs to be `chmod +x`

File: infra/gcp/ensure-audit-perms.sh
Comment (spiffxp) [inline infra/gcp/ensure-audit-perms.sh:43]: For followup, we should either:
- collapse these into a custom role (I'll try to writeup what I did for prow's role, maybe it's time to script composition of predefined roles into a custom role)
- consider granting `roles/viewer` to simplify/avoid brittleness (though this does go against the principle of using most specific permissions possible in production)

**Review by spiffxp — CHANGES_REQUESTED** [review]
This needs changes to run as-is

**Comment by spiffxp** [comment]
In the meantime, to hopefully unblock, I have manually run:
```
gcloud organizations add-iam-policy-binding \
  --member=group:k8s-infra-gcp-auditors@kubernetes.io \
  --role=roles/secretmanager.viewer \
  758905017065
```

File: infra/gcp/ensure-audit-perms.sh
Comment (spiffxp) [reply infra/gcp/ensure-audit-perms.sh:43]: Opened https://github.com/kubernetes/k8s.io/issues/1656 to describe creation of custom role

**Comment by hh** [comment]
Putting on hold for a bit while I focus on getting the audit PR creating prow job working.
/hold

**Comment by spiffxp** [comment]
/close
Closing in favor of https://github.com/kubernetes/k8s.io/pull/1679

**Pull request closed without merging**
